File: backend/src/data/enums.py
from enum import StrEnum


# Define the possible types of items in a survey
class ItemType(StrEnum):
    QUESTION = "question"
    IMAGE = "image"
    VIDEO = "video"
    TEXT = "static_text"  # For static texts or statements


# Define the possible types of questions
class QuestionType(StrEnum):
    FREE_TEXT = "free_text"
    MULTIPLE_CHOICE_SINGLE = "multiple_choice_single"
    MULTIPLE_CHOICE_MULTIPLE = "multiple_choice_multiple"
    SCALE = "scale"
    MATRIX_SCALE = "matrix_scale"
    LIKERT_SCALE = "likert_scale"

File: frontend/src/lib/components/forms/survey/enums.ts
import { CircleHelp, Type, Image, Film } from 'lucide-svelte';

export const ITEM_TYPES = [
	{ value: 'question', label: 'Question', icon: CircleHelp },
	{ value: 'static_text', label: 'Static Text', icon: Type },
	{ value: 'image', label: 'Image', icon: Image },
	{ value: 'video', label: 'Video', icon: Film }
];

export const QUESTION_TYPES = [
	{ value: 'free_text', label: 'Free Text' },
	{ value: 'multiple_choice_single', label: 'Multiple Choice Single' },
	{ value: 'multiple_choice_multiple', label: 'Multiple Choice Multiple' },
	{ value: 'scale', label: 'Scale' },
	{ value: 'matrix_scale', label: 'Matrix Scale' },
	{ value: 'likert_scale', label: 'Likert Scale' }
];
